### PR TITLE
build: bump the minimum python version to 3.9

### PR DESCRIPTION
**Pull request by almet**
Bump the minimum python version to 3.9 when installing from source, and to 3.8 when installing in Debian-related distributions.

The minimum python version when installing from source is now python
3.9, as Pyside6 6.7.1 dropped support for python 3.8 (see https://github.com/freedomofpress/dangerzone/issues/780 for more
information).

On Debian-derivatives distributions, the minimum Python version is now
set to 3.8. In practice, because Pyside6 is not packaged for Debian, we
use Pyside2 [0], which is why we can relax the python version requirement.

In practice, when installing from source on an environment where
python3.9 is not the default python, poetry will look for it and use it
if available

> For various reasons, this Python version might not be compatible with
> the python range supported by the project. In this case, Poetry will
> try to find one that is and use it.
>
> [Poetry docs](https://python-poetry.org/docs/managing-environments/)

On Ubuntu Focal (20.04) where Python 3.9 is not installed by default,
it is possible to install it using the `python3.9` package.

[0] More information on how Pyside6 packaging affects dangerzone on https://github.com/freedomofpress/dangerzone/issues/221
[1] See [the current status of Pyside6 packaging](https://repology.org/project/python:pyside6/packages)

### CODE REVIEW

File: BUILD.md [hunk 1]
```diff
@@ -33,6 +33,28 @@ Install dependencies:
...
 
+
+<table>
+  <tr>
+      <td>
+<details>
+  <summary><i>:memo: Expand this section if you are on Ubuntu 20.04 (Focal).</i></summary>
+  </br>
+
+  The default Python version that ships with Ubuntu Focal is too old for our requirements.
```
[the remark this answers is not in the document]
Comment (almet) [reply BUILD.md]: ✨ I like how this takes out the judgement out of "too old". Thanks for the suggested changes. It's integrated in [`e97a3b8` (#818)](https://github.com/freedomofpress/dangerzone/pull/818/commits/e97a3b8a04a9384b0f087d7a0ae6089fe2b604f6)

File: poetry.lock
[the remark this answers is not in the document]
Comment (almet) [reply poetry.lock:1]: I've just ran `poetry update` and added the changes in [`8f59847` (#818)](https://github.com/freedomofpress/dangerzone/pull/818/commits/8f5984787d9125f2eedae66623d6776666356db7)

[the remark this answers is not in the document]
Comment (apyrgio) [reply poetry.lock:1]: Great, I see that the `requests` package, which Dependabot sent a PR for, gets updated as well.

**Review by apyrgio — APPROVED** [review]
Looks great Alexis! Feel free to merge once the CI tests pass.

File: tests/test_docs/unsafe/sample-bmp.bmp
[the remark this answers is not in the document]
Comment (almet) [reply tests/test_docs/unsafe/sample-bmp.bmp:1]: Yep, I caught this as well and removed it. 😝

**Pull request merged**